src/common/set.h

```diff
@@ -37,7 +37,9 @@
 
 #ifndef NVML_SET_H
 #define NVML_SET_H 1
-
+#ifdef __cplusplus
```

Comment:
This is affecting the library code. I don't understand the reason for this change, but I advise against it.